lcls-twincat-motion/Library/POUs/Motion/FB_MotionHoming.TcPOU

```diff
@@ -0,0 +1,245 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_MotionHoming" Id="{89795143-e01c-4e96-8d50-67b6176895fe}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_MotionHoming
+VAR_IN_OUT
+    stMotionStage: DUT_MotionStage;
+END_VAR
+VAR_INPUT
+    bExecute: BOOL;
+END_VAR
+VAR_OUTPUT
+    bBusy: BOOL;
+    bDone: BOOL;
+    bError: BOOL;
+    nErrorID: UDINT;
+END_VAR
+VAR
+    fbSetPos: MC_SetPosition;
+    fbJog: MC_Jog;
+    rtExec: R_TRIG;
+    ftExec: F_TRIG;
+    nHomeStateMachine: INT := IDLE;
+    nMoves: INT;
+    bFirstDirection: BOOL;
+    bAtHome: BOOL;
+    bMove: BOOL;
+    nErrCount: INT;
+    bInterrupted: BOOL;
+END_VAR
+VAR CONSTANT
+    IDLE: INT := 0;
+    NEXT_MOVE: INT := 1;
+    CHECK_FWD: INT := 2;
+    CHECK_BWD: INT := 3;
+    FINAL_MOVE: INT := 4;
+    FINAL_SETPOS: INT := 5;
+    ERROR: INT := 6;
+
+    FWD_START: LREAL := -99999999;
+    BWD_START: LREAL :=  99999999;
```

Comment:
Quick annotation here: this is a simpler way of disabling the soft limits that ends up being really obvious if something has gone wrong. If you turn the limits off/on, not only do you need to keep track of if you had soft limits set, but you need to always restore this properly or risk some issue.

Instead, I set position to a ridiculous value that can always move forward or backward. If this gets stuck for any reason it's very clear that something has gone wrong, rather than a silent failure of the soft limit marks.

Reply:
(this will become a comment in the code)